### DISCUSSION TITLE
Need some explanations on Git LFS

### DISCUSSION THREAD

**Discussion by TheRedPotato7** [git-lfs/git-lfs#5677] [Q&A]
I am currently trying to push a small unity project to git hub, but it is over 100mb. I have set up lfs just like in the video, but when I push, and I check the git hub page, there are only 2 folders, and the rest of the project is gone. If I try to download and open it with unity, it isn't a proper project, so it doesn't open. If anybody can help, please do.

**Comment by chrisd8088** [comment]
Hey, I'm sorry you're having trouble.  It would be helpful if you provided the output of `git lfs env` run from within your project's Git working tree, at a minimum.  If you can also provide the command line steps you performed, that would be useful as well.  (And if you can link to the video you mention, that might be handy also.)

Typically, if you clone a Git repository with Git LFS pointers, but don't have the Git LFS client fully installed, the pointer files (small regular files which tell Git LFS where to find the actual file data) will be what you find in your working tree, since Git LFS isn't run and so isn't able to populate the working tree with the objects to which the pointers refer.

You might also try `git lfs checkout` in your working tree, as that can sometimes help as well to fully populate the tree with Git LFS objects.

**Reply by TheRedPotato7** [reply]
Im sorry, im new, so I dont know any of those commands. Here are the two videos linked below, which I followed step for step. https://www.youtube.com/watch?v=4Ln6iRh_LTo&t=260s
https://www.youtube.com/watch?v=09McJ2NL7YM
I also ran the command git lfs ls-files, and it returned nothing.

**Reply by chrisd8088** [reply]
The commands I mentioned are all things you would run in a Git Bash terminal window, which is mentioned in the videos.

Are you able to run open such a terminal, change directory into your cloned repository, and run `git lfs env`?  What does that output?

**Reply by TheRedPotato7** [reply]
What part of It do you want? I will just paste the whole thing:
git-lfs/3.4.1 (GitHub; windows amd64; go 1.20.11; git 0898dcbc)
git version 2.44.0.windows.1

Endpoint=https://github.com/TheRedPotato7/RNG.git/info/lfs (auth=basic)
LocalWorkingDir=C:\Users\My name\RNG
LocalGitDir=C:\Users\my name\RNG\.git
LocalGitStorageDir=C:\Users\my name\RNG\.git
LocalMediaDir=C:\Users\my name\RNG\.git\lfs\objects
LocalReferenceDirs=
TempDir=C:\Users\my name\RNG\.git\lfs\tmp
ConcurrentTransfers=8
TusTransfers=false
BasicTransfersOnly=false
SkipDownloadErrors=false
FetchRecentAlways=false
FetchRecentRefsDays=7
FetchRecentCommitsDays=0
FetchRecentRefsIncludeRemotes=true
PruneOffsetDays=3
PruneVerifyRemoteAlways=false
PruneRemoteName=origin
LfsStorageDir=C:\Users\my name\RNG\.git\lfs
AccessDownload=basic
AccessUpload=basic
DownloadTransfers=basic,lfs-standalone-file,ssh
UploadTransfers=basic,lfs-standalone-file,ssh
GIT_EXEC_PATH=C:/Users/my name/AppData/Local/Programs/Git/mingw64/libexec/git-core
git config filter.lfs.process = "git-lfs filter-process"
git config filter.lfs.smudge = "git-lfs smudge -- %f"
git config filter.lfs.clean = "git-lfs clean -- %f"

**Reply by chrisd8088** [reply]
That's good, that means Git LFS is installed and accessible, at least, and the configuration seems reasonable.

Have you tried `git lfs checkout` in the repository?

**Reply by TheRedPotato7** [reply]
Checking out LFS objects: 100% (1/1), 0 B | 0 B/s, done.

**Reply by chrisd8088** [reply]
OK ... do you see the files you want to manage with Git LFS populated in your working tree now?  What do you see with `git lfs ls-files`?  What are you expecting to see?

If you're still having trouble, you'll likely have to provide more detail on what exactly you've done so far, and on the structure of the repository.  For instance, what is in your `.gitattributes` file?  Did you run `git lfs track` as one of your setup steps, and then start adding files to match the filename patterns you defined in that step?

Or did you use `git lfs migrate import` to convert existing files in your existing Git history, and then force-push to the GitHub server?

**Reply by TheRedPotato7** [reply]
When I type the command git lfs ls-files, it returns nothing, but when i didnt have lfs on there was something like 2000 files, but when I did everything, there was 2, and it let me push it.

**Reply by chrisd8088** [reply]
I'm sorry you're having trouble.  You'll have to supply a lot more detail, though, for us to be able to provide any further assistance, particularly about the series of commands you executed.
